cr-examples/triton/src/test/java/oracle/code/triton/TestAddKernel.java

```diff
@@ -105,25 +105,25 @@ public void test(TritonTestData t) {
 
 
     @TritonCodeModel("""
-            module ()void -> {
-                tt.func @"add_kernel2_ptr<float>_ptr<float>_ptr<float>_int_64_void" (%0 : ptr<float>, %1 : ptr<float>, %2 : ptr<float>, %3 : int)void -> {
-                    %4 : int = arith.constant @"64";
-                    %5 : int = tt.get_program_id @"0";
-                    %6 : int = arith.muli %5 %4;
-                    %7 : tensor<x64, int> = tt.make_range @start="0" @end="64";
-                    %8 : tensor<x64, int> = tt.splat %6;
-                    %9 : tensor<x64, int> = arith.addi %8 %7;
-                    %10 : tensor<x64, int> = tt.splat %3;
-                    %11 : tensor<x64, boolean> = arith.cmpi %9 %10 @"slt";
-                    %12 : tensor<x64, ptr<float>> = tt.splat %0;
-                    %13 : tensor<x64, ptr<float>> = tt.addptr %12 %9;
-                    %14 : tensor<x64, float> = tt.load %13 %11;
-                    %15 : tensor<x64, ptr<float>> = tt.splat %1;
-                    %16 : tensor<x64, ptr<float>> = tt.addptr %15 %9;
-                    %17 : tensor<x64, float> = tt.load %16 %11;
-                    %18 : tensor<x64, float> = arith.addf %14 %17;
-                    %19 : tensor<x64, ptr<float>> = tt.splat %2;
-                    %20 : tensor<x64, ptr<float>> = tt.addptr %19 %9;
+            module ()java.type:"void" -> {
+                tt.func @"add_kernel2_ptr<java.type.primitive<float>>_ptr<java.type.primitive<float>>_ptr<java.type.primitive<float>>_int_64_void" (%0 : ptr<java.type:"float">, %1 : ptr<java.type:"float">, %2 : ptr<java.type:"float">, %3 : java.type:"int")java.type:"void" -> {
```

Comment:
Note (in this and other tests) -- the signature of the triton function contains the literal externalized type string of the various parameter/return types. These are NOT flattened strings because e.g. `Ptr::toString` does not flatten (nor it knows how to, if we were to use the public API). This leads to some asymmetry, where in some cases the inflated form appears in full (as here), while in other cases the short type string appears (e.g. because the type is a Java type, and its `toString` is more friendly).

I had half a mind to make `toString` of all triton types more friendly (e.g. not rely on externalize) -- but ultimately I decided to just stick with this very basic conversion -- perhaps we can come back to this later?

Reply:
This is fine IMO. If need be we can revisit later on.